dart/lib/src/scope.dart

```diff
@@ -24,11 +24,10 @@ class Scope {
   ///
   ///     // A completely custom fingerprint:
   ///     var custom = ['foo', 'bar', 'baz'];
-  List<String> get fingerprint =>
-      _fingerprint != null ? List.unmodifiable(_fingerprint) : null;
+  List<String> get fingerprint => List.unmodifiable(_fingerprint);
 
-  set fingerprint(List<String> fingerprint) {
-    _fingerprint = (fingerprint != null ? List.from(fingerprint) : fingerprint);
+  set fingerprint(List<String>? fingerprint) {
+    _fingerprint = List.from(fingerprint ?? []);
```

Comment:
const?
if the field is not nullable, I'd make this param also not nullable, or the other way around